ibis/omniscidb/ddl.py

```diff
@@ -340,6 +341,83 @@ def compile(self):
         return self._wrap_command(action)
 
 
+class AddColumn(AlterTable):
+    """Add Column class."""
+
+    def __init__(self, table_name, **columns_with_types):
+        super().__init__(table_name)
+        self.dict_cols_with_types = {}
+        for key, value in columns_with_types.items():
+            self.dict_cols_with_types[key] = value
+
+    def _pieces(self):
+        col_count = len(self.dict_cols_with_types)
+        if col_count == 0:
+            raise IbisInputError('No column requested to add.')
+        elif col_count == 1:
+            column_name = list(self.dict_cols_with_types.keys())[-1]
+            yield '{} ADD COLUMN {} {};'.format(
+                self.table, column_name, self.dict_cols_with_types[column_name]
+            )
+        else:
+            yield '{} ADD ('.format(self.table)
+            iteration_idx = 0
+            for col, d_type in self.dict_cols_with_types.items():
+                if iteration_idx != col_count - 1:
+                    yield '{} {}, '.format(col, d_type)
+                else:
+                    yield '{} {});'.format(col, d_type)
+                iteration_idx += 1
+
+    def compile(self):
+        """Compile the Add Column expression.
+
+        Returns
+        -------
+        string
+        """
+        cmd = "".join(self._pieces())
+        return self._wrap_command(cmd)
+
+
+class DropColumn(AlterTable):
+    """Drop Column class."""
+
+    def __init__(self, table_name, *column_names):
+        super().__init__(table_name)
+        self.column_names = []
+        for col in column_names:
+            self.column_names.append(col)
+
+    def _pieces(self):
+        col_count = len(self.column_names)
+        if col_count == 0:
+            raise IbisInputError('No column requested to drop.')
+        elif col_count == 1:
+            yield '{} DROP COLUMN {};'.format(
+                self.table, self.column_names[-1]
+            )
+        else:
+            yield '{} '.format(self.table)
+            iteration_idx = 0
+            for col in self.column_names:
+                if iteration_idx != col_count - 1:
+                    yield 'DROP {}, '.format(col)
+                else:
+                    yield 'DROP {};'.format(col)
+                iteration_idx += 1
```

Comment:
you can optimize your code using the same approach that I commented above.

Reply:
Thanks, done